app/models/post.rb

```diff
@@ -80,7 +80,28 @@ class Post < ActiveRecord::Base
   register_custom_field_type(MISSING_UPLOADS_IGNORED, :boolean)
 
   scope :private_posts_for_user, ->(user) {
-    where("posts.topic_id IN (#{Topic::PRIVATE_MESSAGES_SQL})", user_id: user.id)
```

Comment:
It seems a shame to lose the centralised PRIVATE_MESSAGES_SQL... is there some way we can optimize that SQL rather than re-implementing it in two places 🤔

Reply:
The SQL itself isn’t the problem though. It is the use of the subquery that throws the planner off.